src/components/select/simple-select/simple-select.test.tsx

```diff
@@ -1,85 +1,794 @@
+        <Option text="brown" value="brown" />
+        <Option text="green" value="green" />
+      </SimpleSelect>
+    );
+
+    const input = screen.getByRole("combobox");
+    await user.type(input, "bla");
+    act(() => jest.runOnlyPendingTimers());
+    await user.type(input, "g");
+
+    expect(screen.getByText("green", { ignore: "li" })).toBeVisible();
+
+    jest.useRealTimers();
+  });
+
+  it.each(["Meta", "Control"])(
+    "does not select any option when a printable character is typed while holding down the %s key",
+    async (specialKey) => {
+      const user = userEvent.setup();
+      render(
+        <SimpleSelect label="Colour" onChange={() => {}}>
+          <Option text="amber" value="amber" />
+          <Option text="blue" value="blue" />
+          <Option text="black" value="black" />
+        </SimpleSelect>
+      );
+
+      // Hold special key down while typing 'b'
+      await user.type(screen.getByRole("combobox"), `{${specialKey}>}b`);
+
+      expect(screen.getByText("Please Select...")).toBeVisible();
+    }
+  );
+
+  it("does not change selected option when value prop is set (controlled usage)", async () => {
+    const user = userEvent.setup();
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} value="amber">
+        <Option text="amber" value="amber" />
+        <Option text="blue" value="blue" />
+      </SimpleSelect>
+    );
+
+    await user.type(screen.getByRole("combobox"), "blue");
+
+    expect(
+      screen.queryByText("blue", { ignore: "li" })
+    ).not.toBeInTheDocument();
+  });
+
+  it("calls onChange callback each time a character typed into the input", async () => {
+    const user = userEvent.setup();
+    const onChange = jest.fn();
+    render(
+      <SimpleSelect label="Colour" onChange={onChange}>
+        <Option text="blue" value="blue" />
+        <Option text="black" value="black" />
+        <Option text="brown" value="brown" />
+      </SimpleSelect>
+    );
+
+    await user.type(screen.getByRole("combobox"), "bla");
+
+    expect(onChange).toHaveBeenCalledTimes(3);
+  });
+
+  it("calls onChange callback with the value of the matched option after a character is typed into the input", async () => {
+    const user = userEvent.setup();
+    const onChange = jest.fn();
+    render(
       <SimpleSelect
-        aria-labelledby="my-select-heading"
+        label="Colour"
+        name="colour"
+        id="colour"
+        onChange={onChange}
+      >
+        <Option text="blue" value="blue" />
+        <Option text="black" value="black" />
+        <Option text="brown" value="brown" />
+      </SimpleSelect>
+    );
+
+    await user.type(screen.getByRole("combobox"), "b");
+
+    expect(onChange).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        target: { value: "blue", name: "colour", id: "colour" },
+        selectionConfirmed: false,
+      })
+    );
+  });
+});
+
+describe("opening the dropdown list", () => {
```

Comment:
suggestion (non-blocking): these don't really read out correctly, when they run it will be `opening the dropdown list -> opens list when input is clicked` etc. Maybe something like below would work:

```
describe("dropdown list", () => {
it("opens when input is clicked", async () => {
...
it("does not open when input is disabled and is clicked", async () => {

Reply:
You could then include the closing list stuff in the same describe etc

```
it("closes when open and an option is clicked", async () => {